HealthMonitoring.AcceptanceTests/Scenarios/Selenium/Home_page.Steps.cs

```diff
@@ -186,6 +188,48 @@ public void Then_only_endpoints_with_chosen_parameters_should_be_shown()
             Assert.True(selectedTags.Any(m => string.Equals(m.Text, tag)));
         }
 
+        public void With_group_filter_input(string groupFilterValue)
+        {
+            var groupFilter = _driver.FindElement(By.Id("groupFilter"));
+
+            groupFilter.SendKeys(groupFilterValue);
+        }
+
+        public void With_name_filter_input(string nameFilterValue)
+        {
+            var nameFilter = _driver.FindElement(By.Id("nameFilter"));
+
+            nameFilter.SendKeys(nameFilterValue);
+        }
+
+        public void Then_only_endpoints_with_groupfilter_parameters_should_be_shown(string groupFilterValue)
+        {
+            IList<IWebElement> groupElements = _driver
+                .WaitElementsAreRendered(
+                    By.XPath(_endpointsGroupsSelectorValue), 
+                    group => group.Displayed && group.Enabled)
+                .ToList();
+
+            bool areStatusesCorrect = groupElements.All(m => m.Text.ToLower().IndexOf(groupFilterValue, StringComparison.OrdinalIgnoreCase) >= 0);
+
+            Assert.True(areStatusesCorrect);
+            Assert.True(groupElements.Count(m => m.Text.ToLower().IndexOf(groupFilterValue, StringComparison.OrdinalIgnoreCase) >= 0) == groupElements.Count);
```

Comment:
there is extension method (in Linq): m.Text.Contains("sdf", StringComparer.OrdinalIgnoreCase))
that takes comparer as 2 parameter